src/Synchronizer.php

```diff
@@ -148,22 +156,67 @@ private function getHostsLines($container)
     }
 
     /**
-     * @param Container $container
+     * @param array $container
+     *
+     * @return string
+     */
+    private function getProjectName($container)
+    {
+        if (isset($container['Config']['Labels']['com.docker.compose.project'])) {
+            return $container['Config']['Labels']['com.docker.compose.project'];
+        }
+
+        return '';
+    }
+
+    /**
+     * @param array $container
      *
      * @return array
      */
     private function getContainerHosts($container)
     {
-        $hosts = [substr($container['Name'], 1).$this->tld];
+        return array_merge(
+            [substr($container['Name'], 1).$this->tld],
+            $this->getAdditionalContainerHosts($container)
+        );
+    }
+
+    /**
+     * @param array $container
+     *
+     * @return array
+     */
+    private function getAdditionalContainerHosts($container)
+    {
+        $hosts = [];
+
+        $domainName = $this->getEnv($container, 'DOMAIN_NAME');
```

Comment:
I would like to suggest that `DOMAIN_NAME` is not entirely correct name for what we want to achieve with this feature. "Domain" is rather HTTP concept, but host manager can be used also for database containers instead of port binding (so developer can connect to database with client by custom host, not by `localhost` with bound port).